src/modules/Filter/Filter.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Stack, Typography, Box, Button, IconButton, Popover, Checkbox } from '@mui/material';
+import FilterListIcon from '@mui/icons-material/FilterList';
+import Close from '@mui/icons-material/Close';
+import Accordion from '../../components/Accordian/Accordian';
+import IFilterGroup from '../../types/FilterGroup';
+
+interface IProps {
+  filters: IFilterGroup[];
+}
+
+const Filter: React.FC<IProps> = ({ filters }) => {
+  const [expanded, setExpanded] = React.useState<number[]>(filters.map((data) => data.id));
+  const handleChange = (panelId: number) => () => {
+    if (expanded.includes(panelId)) setExpanded(expanded.filter((data) => data !== panelId));
+    else setExpanded(expanded.concat(panelId));
+  };
+  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  const filterDetails = filters.map((filterGroup) => (
+    <Accordion
+      title={filterGroup.type}
+      expanded={expanded.includes(filterGroup.id)}
+      onChange={handleChange(filterGroup.id)}
+    >
+      {filterGroup.filters.map((filter) => (
+        <Stack direction="row" alignItems="center" paddingX={1.375} paddingY={1}>
+          <Checkbox />
+          <Typography variant="body2">{filter}</Typography>
+        </Stack>
+      ))}
+    </Accordion>
+  ));
+
+  return (
+    <>
+      <Box display="flex" alignItems="flex-end" flexDirection="column">
+        <Button disableRipple variant="text" onClick={handleClick} sx={{ textTransform: 'none' }}>
+          <Stack direction="row" alignItems="center" gap={1.5}>
+            <FilterListIcon />
+            <Typography variant="subtitle1" fontWeight={500}>
+              Filter
+            </Typography>
+          </Stack>
+        </Button>
+      </Box>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        PaperProps={{
+          style: { border: '1px solid #C5C5C5', boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.25)', borderRadius: '6px' },
+        }}
+      >
+        <>
+          <Box
+            display="flex"
+            justifyContent="space-between"
+            paddingX={2.5}
+            paddingTop={2.5}
+            paddingBottom={1.25}
```

Comment:
these can be replaced with px,pt,pb. fine for now. but try to use these, as it's simple to use them.